internal/validate/validate.go

```diff
@@ -58,6 +58,14 @@ func Time(source string, t time.Time) []shared.FieldError {
 	return If(t.IsZero(), []shared.FieldError{{Source: source, Detail: "field is required"}})
 }
 
+func OptionalTime(source string, t *time.Time) []shared.FieldError {
+	if t == nil {
+		return nil
+	}
+
+	return If(t.IsZero(), []shared.FieldError{{Source: source, Detail: "field is required"}})
```

Comment:
This feels a bit misleading, since the field isn't actually required. Could we use something like "field must be a valid date/time"?